blog/_posts/2016-05-17-automated-data-validation.md

```diff
@@ -0,0 +1,136 @@
+the current status of the project's most recent build at a glance.
+
+![Build Passing](/img/posts/build-passing.png)
+
+As with software, datasets are often collaboratively created, edited,
+and updated over time, sometimes introducing subtle (or not so subtle)
+structural and schematic errors (see
+[Bad Data](http://okfnlabs.org/bad-data/) for examples).  Much of the
+"friction" in working with data comes from the time and effort needed
+to identify and address these errors before analyzing in a given tool.
+Automatically flagging *data* quality issues at upload time in a
+repository can go a long way in making data more useful and have
+significant follow-on effects in the data ecosystem, both open and
+closed.
+
+## Continuous Data Integration
+
+As the [Frictionless Data](http://frictionlessdata.io/) tooling and
+standards ecosystem continues to grow, we now have the elements
+necessary to provide data managers with the same type of service for
+tabular data (e.g. Excel and CSV).  In less than one hour, a few of us
+at Open Knowledge booted a small demo to show what *continuous data
+integration* could look like.  On each commit to
+[our example repository](https://github.com/frictionlessdata/ex-continuous-data-integration),
+a set of validation tests are run on the data, raising an exception if
+the data is invalid.  If a user adds "bad" data, the "build" fails and
+issues a report indicating what went wrong.
+
+[![Data CI](/img/posts/data_ci_travis.png)](https://github.com/frictionlessdata/ex-continuous-data-integration)
+
+As an example, this CSV is "bad".
+
+```
+Date,Country,Number
+2015-01-01,3,20.3
+2015-02-01,United States,23.5
+2015-02,United States,x23.5
+```
+
+When we packaged it (see its
+[datapackage.json](https://github.com/frictionlessdata/ex-continuous-data-integration/blob/master/datapackage.json)),
+we gave it the following JSON Table Schema.
+
+{% highlight json %}
+{
+  "fields": [
+    {
+      "name": "Date",
+      "type": "date",
+      "description": ""
+    },
+    {
+      "name": "Country",
+      "type": "string",
+      "description": ""
+    },
+    {
+      "name": "Number",
+      "type": "number",
+      "description": ""
+    }
+  ]
+}
+{% endhighlight %}
+
+On commit, this data would result in the following report:
+
+```
++----------------+-------------------------------------------------------------+
+| result_name    | result_message                                              |
++================+=============================================================+
+| Incorrect Type | The value "2015-02" in column "Date" is not a valid Date.   |
++----------------+-------------------------------------------------------------+
+| Incorrect Type | The value "x23.5" in column "Number" is not a valid Number. |
++----------------+-------------------------------------------------------------+
+```
+
+## How It Works
+
+The Data Package descriptor file,
```

Comment:
Worth inlining an excerpted version of the script with comments ...

Screenshot of the datapackage.json with the schema highlighted within it?